lib/util/SortableSet.js

```diff
@@ -1,17 +1,30 @@
 "use strict";
 
+/**
+ * A subset of Set that offers sorting functionality
+ * @template T
+ * @typedef {(a: T, b: T) => number} SortFn
```

Comment:
Put `@typedef` into separate comment.

Reply:
This has to be here because it uses the generic. Templates in classes are not working yet though:

https://github.com/Microsoft/TypeScript/issues/23459

Reply:
Support for `@template` on classes will be in `typescript@next` starting tonight. The typedef here will still work like it's local to the class. But that is confusing so I would recommend moving it inside the class.